### PR TITLE
Implemented upvote button to vote on articles

### PR DESCRIPTION
**Pull request by Tiny-B**
This pr implements functionality to the upvote button on the articles, it can be pressed to add a vote and pressed again to remove the vote, the state of the button is stored in the browsers local storage using the article_id as an identifier

### CODE REVIEW

File: src/components/article-components/FullArticle.jsx [hunk 1]
```diff
@@ -1,4 +1,46 @@
...
+		if (hasVotedFromStorage) {
+			setHasVoted(JSON.parse(hasVotedFromStorage));
+		}
+	}, [article.article_id]);
+
+	useEffect(() => {
+		localStorage.setItem(
+			`hasVoted-${article.article_id}`,
+			JSON.stringify(hasVoted)
+		);
```
Comment (HeatherEver) [inline src/components/article-components/FullArticle.jsx:41]: this is an interesting way to approach this but using localStorage isn't the best way to go about this - i would think about how you could refactor this at a later date

File: src/components/article-components/FullArticle.jsx [hunk 2]
```diff
@@ -1,4 +1,46 @@
...
+
+	const handleUpvoteButton = e => {
+		api
+			.upvoteArticle(article.article_id, e.target.value)
+			.then(() => {
+				setVotes(current => {
+					return Number(current) + Number(e.target.value);
+				});
+
+				if (e.target.value == 1) {
```
Comment (HeatherEver) [inline src/components/article-components/FullArticle.jsx:16]: instead of using double equals here you could strictly compare the string of 1

**Review by HeatherEver — APPROVED** [review]
Nice work Kief and i love the way the vote button changes when clicked and nice alert when the user is offline 🎉  
However, you're not optimistically rendering the votes so if there's slow internet the user has to wait for the patch request to go through before it will update so please can this be changed :)

File: src/components/article-components/FullArticle.jsx (hunk 1 above)
Comment (Tiny-B) [reply src/components/article-components/FullArticle.jsx:41]: Absolutely agree

**Pull request merged**